scandium-core/src/main/java/org/eclipse/californium/scandium/auth/PrincipalSerializer.java

```diff
@@ -91,7 +91,10 @@ public static void serialize(final Principal principal, final DatagramWriter wri
 
 	private static void serializeIdentity(final PreSharedKeyIdentity principal, final DatagramWriter writer) {
 		writer.writeByte(ClientAuthenticationType.PSK.code);
-		writeBytesWithLength(PSK_LENGTH_BITS, principal.getName().getBytes(StandardCharsets.UTF_8), writer);
+		byte[] virtualHost = principal.getVirtualHost() == null ? new byte[0] :
```

Comment:
I would prefer the have a `static final byte[] EMPTY = {};` which could be reused.